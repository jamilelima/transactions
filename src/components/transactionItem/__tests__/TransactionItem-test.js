import React from 'react';
import TransactionItem from '../index';
import {create} from 'react-test-renderer';

describe('Test TransactionItem component', () => {
  const transaction = {
    id: 0,
    description: 'Lorem ipsum mock data',
    value: 100.0,
    type: 'debit',
    date: '22/09/1996',
  };

  const transactionItem = create(<TransactionItem data={transaction} />);
  test('Match with snapashot', () => {
    expect(transactionItem.toJSON()).toMatchSnapshot();
  });

  test('Test data prop', () => {
    const instance = transactionItem.root;
    expect(instance.props.data).toBe(transaction);
  });
});
