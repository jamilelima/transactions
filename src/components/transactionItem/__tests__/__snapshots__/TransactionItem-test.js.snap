// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TransactionItem component Match with snapashot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#1B232F",
        "borderRadius": 10,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
        "paddingBottom": 30,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 30,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#F76353",
          "fontSize": 24,
        },
        undefined,
        Object {
          "fontFamily": "Feather",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ÔÑè
  </Text>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontFamily": "Nunito-Regular",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          },
        ]
      }
    >
      Lorem ipsum mock data
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#90949c",
            "fontFamily": "Nunito-Regular",
            "fontSize": 12,
            "textAlign": "center",
          },
        ]
      }
    >
      22/09/1996
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Nunito-Regular",
          "fontSize": 16,
          "fontWeight": "bold",
        },
        Object {
          "color": "#F76353",
        },
      ]
    }
  >
    R$ 
    100,00
  </Text>
</View>
`;
